contentcuration/contentcuration/tests/test_contentnodes.py

```diff
@@ -171,6 +173,47 @@ def test_get_node_details(self):
         assert details["resource_size"] > 0
         assert len(details["kind_count"]) > 0
 
+    def test_get_tree_id_by_pk(self):
+        from django.conf import settings
+
+        contentnode = ContentNode.objects.create(
+            title="Kolibri", parent=self.channel.main_tree, kind=self.topic
+        )
+
+        assert settings.IS_CONTENTNODE_TABLE_PARTITIONED is False
+        node = ContentNode.filter_by_pk(contentnode.id)[0]
+        tree_id_from_cache = cache.get(CONTENTNODE_TREE_ID_CACHE_KEY.format(pk=contentnode.pk))
+        assert node.id == contentnode.id
+        assert tree_id_from_cache is None
+        with patch("contentcuration.models.ContentNode") as mock_contentnode:
+            ContentNode.filter_by_pk(contentnode.id)
+            mock_contentnode.objects.filter.assert_called_once_with(pk=contentnode.id)
+
+        settings.IS_CONTENTNODE_TABLE_PARTITIONED = True
```

Comment:
You should consider splitting this test into individual methods. Also, it's a good idea to use `override_settings` here https://docs.djangoproject.com/en/3.0/topics/testing/tools/#django.test.override_settings

This method tests the happy path of when the PK exists, but doesn't test the case of when the PK doesn't exist. Adding additional test methods for those cases is a good idea.